File: frontend/src/types/recommendation.js
/**
 * @typedef {Object} Recommendation
 * @property {string} actionCode
 * @property {string} actionName
 * @property {number} score
 * @property {Object} scoreBreakdown
 * @property {number} scoreBreakdown.transitionScore
 * @property {number} scoreBreakdown.knowledgeScore
 * @property {number} scoreBreakdown.growthStageScore
 * @property {number} scoreBreakdown.weatherSuitabilityScore
 * @property {number} scoreBreakdown.symptomUrgencyScore
 * @property {number} scoreBreakdown.recencyPenaltyScore
 * @property {number} scoreBreakdown.finalScore
 * @property {number} expectedDay
 * @property {string} reason
 * @property {string[]} reasonItems
 * @property {string[]} candidateSources
 * @property {string} sourceType
 */
export {}

File: backend/app/models/recommendation.py
from datetime import datetime

from sqlalchemy import DateTime, Float, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database.base import Base


class Recommendation(Base):
    __tablename__ = "recommendations"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    taskId: Mapped[str] = mapped_column(ForeignKey("agent_tasks.id", ondelete="CASCADE"), index=True)
    actionCode: Mapped[str] = mapped_column(String(50))
    actionName: Mapped[str] = mapped_column(String(100))
    score: Mapped[float] = mapped_column(Float)
    reason: Mapped[str] = mapped_column(Text)
    scoreBreakdown: Mapped[str] = mapped_column(Text, default="{}")
    reasonItems: Mapped[str] = mapped_column(Text, default="[]")
    candidateSources: Mapped[str] = mapped_column(Text, default="[]")
    expectedDay: Mapped[int] = mapped_column(Integer)
    sourceType: Mapped[str] = mapped_column(String(50))
    createdAt: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

    task = relationship("AgentTask", back_populates="recommendations")
